docs/sdk/serverless-signing/zero-to-hero.md

```diff
@@ -0,0 +1,181 @@
+
+const nonce = litNodeClient.getLatestBlockhash();
+```
+
+</TabItem>
+
+<TabItem value="nodejs">
+
+```jsx
+import * as LitJsSdk from "@lit-protocol/lit-node-client-nodejs";
+
+const litNodeClient = new LitJsSdk.LitNodeClientNodeJs({ litNetwork: 'cayenne' });
+await litNodeClient.connect();
+
+const nonce = litNodeClient.getLatestBlockhash();
+```
+
+</TabItem>
+</Tabs>
+
+**Note:** The `mintInfo` contains all the required info for the PKP including its `tokenId` & `publicKey`.
+
+```jsx
+const mintCost = await contractClient.pkpNftContract.read.mintCost(); you can check how much it costs to mint a PKP with this
+
+const authMethod = {
+  authMethodType: AuthMethodType.EthWallet,
+  accessToken: JSON.stringify(authSig),
+};
+
+const mintInfo = await contractClient.mintWithAuth({
+  authMethod,
+  scopes: [
+    AuthMethodScope.NoPermissions,
+    AuthMethodScope.SignAnything,
+    AuthMethodScope.OnlySignMessages,
+  ],
+});
+```
+
+## Check the scope for the PKP
+
+This step isn't necessary for signing with the PKP but can be done to view whether the minted PKP has proper scopes which are required for signing. The first step is to get the `authId` from the `authMethod` as the PKP contracts stores a mapping from the `authId` & its scopes. The `3` below is just the maxScopeId which should be greater than the number of Auth Method scopes that you define.
+
+```jsx
+const authId = await LitAuthClient.getAuthIdByAuthMethod(authMethod);
+await contractClient.pkpPermissionsContract.read.getPermittedAuthMethodScopes(
+  mintInfo.pkp.tokenId,
+  AuthMethodType.EthWallet,
+  authId,
+  3
+);
+
+const signAnythingScope = scopes[1];
+const onlySignMessagesScope = scopes[2];
+```
+
+## Lit Action Signing with the PKP
+
+We'll again use the `litNodeClient` to call the `executeJs` to sign the message with the PKP.
+
+:::note
+`toSign` data is required to be in 32 byte format. 
+
+The `ethers.utils.arrayify(ethers.utils.keccak256(...)` can be used to convert the `toSign` data to the correct format.
+:::
+
+Set up the Lit Action code to be run on the Lit nodes.
+
+```jsx
+const litActionCode = `
+    const go = async () => {
+    // The params toSign, publicKey, sigName are passed from the jsParams fields and are available here
+    const sigShare = await Lit.Actions.signEcdsa({ toSign, publicKey, sigName });
+    };
+
+    go();
+`;
+
+const signatures = await litNodeClient.executeJs({
+  code: litActionCode,
+  authSig,
+  jsParams: {
+    toSign: [84, 104, 105, 115, 32, 109, 101, 115, 115, 97, 103, 101, 32, 105, 115, 32, 101, 120, 97, 99, 116, 108, 121, 32, 51, 50, 32, 98, 121, 116, 101, 115],
+    publicKey: mintInfo.pkp.publicKey,
+    sigName: "sig1",
+  },
+});
+
+console.log("signatures: ", signatures);
+```
+
+## Conclusion
+
+The signatures above are the signatures from the nodes using the PKP. In Cayenne we have 3 nodes so only 2 valid signature shares are required to combine the shares. Hence you will see one od the node always fail to sign.
```

Comment:
i think we should add this as a note in the section above ( Lit Action Signing with the PKP)

for the conclusion, let's re-emphasize what the section was about ('this section showed you how you can mint a PKP and use it to sign messages with Lit Actions. To learn more, check out these resources: 

- [Generating signed transactions](https://developer.litprotocol.com/v3/sdk/serverless-signing/processing-validation)
- [Fetching off-chain data](https://developer.litprotocol.com/v3/sdk/serverless-signing/fetch)
- [Connecting PKPs to dApps](https://developer.litprotocol.com/v3/sdk/wallets/walletconnect)